rlImGui.cpp

```diff
@@ -282,7 +282,20 @@ static void rlImGuiRenderTriangles(unsigned int count, int indexStart, const ImV
 static void EnableScissor(float x, float y, float width, float height)
 {
     rlEnableScissorTest();
-    rlScissor((int)x, GetScreenHeight() - (int)(y + height), (int)width, (int)height);
+#if defined(__APPLE__)
+    Vector2 scale = GetWindowScaleDPI();
+    rlScissor((int)(x*scale.x), (int)(GetScreenHeight()*scale.y - (((y + height)*scale.y))), (int)(width*scale.x), (int)(height*scale.y));
+#else
+    if ((CORE.Window.flags & FLAG_WINDOW_HIGHDPI) > 0)
```

Comment:
How does this file get the CORE structure? That is part of rcore.c and not in any header. How does the compiler know when building this on apple what is in the CORE structure?

Reply:
The __APPLE__ define is provided by Apple SDK, and is available when compiling for Apple. You make a good point about a dependency on CORE from rlgl. I'm not an expert on Raylib, so perhaps there is a better strategy for getting window scale DPI in rlgl, or a change is required in rlgl to expose that information in the rlgl API (as it is available below rlgl). I'm not entirely happy with using the __APPLE__ define, because it assumes that only Apple has window scaling DPI other than 1.0, which I'm not sure is the case. The reason the fix [Fix scissor on macos #2170](https://github.com/raysan5/raylib/pull/2170)) was done like this, was because the problem has only been found on a MacBook Pro (to my knowledge), and less risk in breaking non-Apple platforms.

Reply:
Oh, the Apple #def I fully understand, I have no problem with that, I've used it before. The use of CORE as an unexposed struct it what I have issue with. If raylib were to change that internal structure this code would break. I'm not even sure how it compiles, the compiler can't know what the structure is without an include. I am concerned that this is only working due to clang/llvm undefined behavior that is fragile.

Reply:
Looking at my build, the reason it compiles is that I am exposing Raylib and rlImGui in my cmake script (in hindsight an error). I'm using cmake instead of premake. If you are interested in the details you can see my [fork with cmake branch](https://github.com/lukekras/rlImGui/blob/add_cmake/CMakeLists.txt). The more I look at this, the more I see that my PR is not ready for a merge into master. Certainly there is an issue on MacBook Pros (because of their window scale DPI of 2.0).

Reply:
Cmake doesn't explain how the compiler is ok with knowing the structure of the CORE global in rlImGui.cpp. When compiling(not linking) rlImGui.cpp, how does it know what fields are in CORE? It is not exposed in any header file, and no file that is included by the ImGui cpp file. Did you expose CORE in a modified raylib header or something? Cmake just sets up the build, I can't see how this compiles once that build is setup. I don't see this as a premake vs cmake thing but just a general compile question.

Reply:
I've only been running this on a MacBook Pro and so my compiler preprocessor will strip out the code with CORE in it because of the __ APPLE __ define. So that should explain why mine is compiling.